lang/py/test/test_io.py

```diff
@@ -38,6 +38,21 @@
   ('"float"', 1234.0),
   ('"double"', 1234.0),
   ('{"type": "fixed", "name": "Test", "size": 1}', 'B'),
+  ('{"type": "int", "logicalType": "date"}', datetime.date(2000, 1, 1)),
+  ('{"type": "int", "logicalType": "time-millis"}', datetime.time(2, 2, 1, 123000)),
+  ('{"type": "int", "logicalType": "time-millis"}', datetime.time(2, 2, 1, 120000)),
+  ('{"type": "long", "logicalType": "time-micros"}', datetime.time(2, 2, 1, 123456)),
+  ('{"type": "long", "logicalType": "time-micros"}', datetime.time(2, 2, 1, 123450)),
+  ('{"type": "long", "logicalType": "timestamp-millis"}', datetime.datetime(2000, 1, 18, 2, 2, 1, 123000, tzinfo=pytz.utc)),
+  (
+    '{"type": "long", "logicalType": "timestamp-millis"}',
+    datetime.datetime(2000, 1, 18, 2, 2, 1, 100000, tzinfo=pytz.timezone('Europe/Moscow'))
+  ),
```

Comment:
It's probably also worth adding a test case that the datetime is before 1970/1/1.